### PR TITLE
feat(dws): add datasource to get logical clusters

### PR DESCRIPTION
**Pull request by wuzhuanhong**
**What this PR does / why we need it**:

Add new data source to get the list of the DWS logical clusters.

In some region, such as **cn-sourth-1**, when the cluster ID does not exist, the response body is in base64 format. It will be automatically converted to the standard error format in the **provider_client** file of the project framework, and no additional processing is required.
image

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add new data source.
2. add related document and acceptance test.
```

## PR Checklist

* [x] Tests added/passed.

```
make testacc TEST='./huaweicloud/services/acceptance/dws' TESTARGS='-run TestAccDataSourceLogicalClusters_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dws -v -run TestAccDataSourceLogicalClusters_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceLogicalClusters_basic
=== PAUSE TestAccDataSourceLogicalClusters_basic
=== CONT  TestAccDataSourceLogicalClusters_basic
--- PASS: TestAccDataSourceLogicalClusters_basic (243.36s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dws       243.403s
```

* [x] Documentation updated.
* [x] Schema updated.
* [ ] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

    
  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

### CODE REVIEW

File: docs/data-sources/dws_logical_clusters.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String) Specifies the region in which to query the resource.
+  If omitted, the provider-level region will be used.
+
+* `cluster_id` - (Required, String) Specified the cluster ID of the DWS.
```
Comment (Lance52259) [inline docs/data-sources/dws_logical_clusters.md]:
```suggestion
* `cluster_id` - (Required, String) Specified the ID of the cluster to which the logical clusters belong.
```

Comment (wuzhuanhong) [reply docs/data-sources/dws_logical_clusters.md]: fixed

**Comment by wuzhuanhong** [comment]
make testacc TEST='./huaweicloud/services/acceptance/dws' TESTARGS='-run TestAccDataSourceLogicalClusters_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dws -v -run TestAccDataSourceLogicalClusters_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceLogicalClusters_basic
=== PAUSE TestAccDataSourceLogicalClusters_basic
=== CONT  TestAccDataSourceLogicalClusters_basic
--- PASS: TestAccDataSourceLogicalClusters_basic (179.81s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dws       179.925s

**Pull request merged**